runtime/manifest.js

```diff
@@ -271,26 +271,35 @@ class Manifest {
           slotConn = particle.addSlotConnection(slotConnectionItem.param);
         }
         if (slotConnectionItem.name) {
-          let targetSlot = items.byName.get(slotConnectionItem.name);
-          if (!targetSlot || !items.bySlot.has(targetSlot)) {
-            targetSlot = recipe.newSlot();
-            targetSlot.localName = slotConnectionItem.name;
-            items.byName.set(slotConnectionItem.name, targetSlot);
-          }
-          slotConn.connectToSlot(targetSlot);
-
           slotConnectionItem.providedSlots.forEach(ps => {
-            let providedSlot = items.byName.get(ps.name) || slotConn.providedSlots[ps.param];
+            let providedSlot = slotConn.providedSlots[ps.param];
+            if (providedSlot) {
+              items.byName.set(ps.name, providedSlot);
+              items.bySlot.set(providedSlot);
```

Comment:
bySlot is a map, doesn't it need set(key, value)?
and same items.bySlot.add(...) below